### PR TITLE
Fix scheduler crashes when profiling is enabled

### LINKED ISSUE #6621: Scheduler is crashing when profiling is switched on
panic: http: multiple registrations for /debug/pprof/
goroutine 11 [running]:
net/http.(ServeMux).Handle(0xc20800a8a0, 0x9e9490, 0xd, 0x7f7d12f190a0, 0xb0e7c0)
/usr/src/go/src/net/http/server.go:1557 +0x244
net/http.(ServeMux).HandleFunc(0xc20800a8a0, 0x9e9490, 0xd, 0xb0e7c0)
/usr/src/go/src/net/http/server.go:1585 +0x6d
net/http.HandleFunc(0x9e9490, 0xd, 0xb0e7c0)
/usr/src/go/src/net/http/server.go:1597 +0x48
github.com/GoogleCloudPlatform/kubernetes/plugin/cmd/kube-scheduler/app.func·001()
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/plugin/cmd/kube-scheduler/app/server.go:84 +0x58
created by github.com/GoogleCloudPlatform/kubernetes/plugin/cmd/kube-scheduler/app.(*SchedulerServer).Run
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/plugin/cmd/kube-scheduler/app/server.go:90 +0x1c7

### PR DESCRIPTION
**Pull request by wojtek-t**
Fixes #6621 

cc @fgrzadkowski @timothysc

### CODE REVIEW

File: plugin/cmd/kube-scheduler/app/server.go [hunk 1]
```diff
@@ -80,13 +81,22 @@ func (s *SchedulerServer) Run(_ []string) error {
 	}
 
 	go func() {
+		mux := http.NewServeMux()
 		if s.EnableProfiling {
```
Comment (timothysc) [inline plugin/cmd/kube-scheduler/app/server.go]: Should we just enable by default?  Then we can capture and report data in e2e.  I know @vmarmol thought so.

Comment (wojtek-t) [inline plugin/cmd/kube-scheduler/app/server.go]: I'm fine with enabling it by default, but I think it should be consistent among all component (i.e. we should also enable it in apiserver and controller). What do you think?

Comment (vmarmol) [inline plugin/cmd/kube-scheduler/app/server.go]: FWIW, there shouldn't be an overhead to having this unless it is used (as in, queried). One downside is that exposing it publicly could allow an avenue for DOSing through it. I don't think that's a big concern today though.

File: plugin/cmd/kube-scheduler/app/server.go [hunk 2]
```diff
@@ -80,13 +81,22 @@ func (s *SchedulerServer) Run(_ []string) error {
...
+			mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
 		}
-		http.Handle("/metrics", prometheus.Handler())
-		http.ListenAndServe(net.JoinHostPort(s.Address.String(), strconv.Itoa(s.Port)), nil)
+		mux.Handle("/metrics", prometheus.Handler())
+
+		server := &http.Server{
+			Addr:           net.JoinHostPort(s.Address.String(), strconv.Itoa(s.Port)),
+			Handler:        mux,
+			ReadTimeout:    5 * time.Minute,
```
Comment (vmarmol) [inline plugin/cmd/kube-scheduler/app/server.go]: Why are we setting these three values btw?

Comment (wojtek-t) [inline plugin/cmd/kube-scheduler/app/server.go]: Those timeouts are probably unnecessary. However, the Addr & Handler were necessary (I mean I didn't know how to make both debug & metrics handlers). Will fix it.

Comment (vmarmol) [inline plugin/cmd/kube-scheduler/app/server.go]: Yeah I was wondering about the timeouts and the max header.

File: plugin/cmd/kube-scheduler/app/server.go (hunk 1 above)
Comment (timothysc) [inline plugin/cmd/kube-scheduler/app/server.go]: I'll get a PR for the other daemons.

File: plugin/cmd/kube-scheduler/app/server.go (hunk 2 above)
Comment (wojtek-t) [inline plugin/cmd/kube-scheduler/app/server.go]: Done.

**Comment by timothysc** [comment]
LGTM, now.

**Comment by vmarmol** [comment]
LGTM, will wait for green.

**Comment by vmarmol** [comment]
CI failure:

```
# github.com/GoogleCloudPlatform/kubernetes/plugin/cmd/kube-scheduler/app
_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/plugin/cmd/kube-scheduler/app/server.go:28: imported and not used: "time"
```

**Comment by wojtek-t** [comment]
Sorry - will fix that in a minute.

**Comment by vmarmol** [comment]
Travis is green!

**Pull request merged**